docs/diagrams/ParserClasses.puml

```diff
@@ -19,20 +19,21 @@ Class Prefix
 }
 
 Class HiddenOutside #FFFFFF
-HiddenOutside ..> AddressBookParser
+HiddenOutside ..> NuSchedulerParser
 
-AddressBookParser .down.> XYZCommandParser: creates >
-
-XYZCommandParser ..> XYZCommand : creates >
-AddressBookParser ..> Command : returns >
-XYZCommandParser .up.|> Parser
-XYZCommandParser ..> ArgumentMultimap
-XYZCommandParser ..> ArgumentTokenizer
+NuSchedulerParser .down.> XCommandParser: creates >
```

Comment:
a ZXCommandParser should inherit from XCommandParser

Reply:
I checked the code again and I think the `ZXCommandParser` also implements the `Parser` interface using generic type of `ZXCommand`. But for the `ZXCommand`, it does inherit `XCommand`. I will update the mistakes in the diagrams and DG, thanks for catching!